packages/metals-vscode/package.json

```diff
@@ -250,8 +250,14 @@
           "markdownDescription": "List of packages you'd like to be left out of completions, symbol searches, and code actions.\n\nEx. `akka.actor.typed.javadsl` will ensure nothing in the `javadsl` package gets recommended to you.\n\nYou can find the list of default exclusions [here on the Metals website](https://scalameta.org/metals/docs/editors/user-configuration/#excluded-packages).\n\nIf you need to remove one of the defaults, you can simply include it and preface it with `--`."
         },
         "metals.showInferredType": {
-          "type": "boolean",
-          "markdownDescription": "When this option is enabled, for each method that have inferred types that are not explicitely specified they are displayed as additional decorations."
+          "type": "string",
+          "default": "false",
+          "enum": [
+            "false",
+            "true",
+            "minimal"
```

Comment:
aren't `false` and `true` a bit unclear if value is an enum? Maybe `off`, `minimal`, `all` would be better? Although I guess this will be slightly incompatible change on the server side. I'm not insisting, but I'd like to hear some opinions before merging this PR.

Reply:
Sure this would make sense, but we would need to change the configuration on the Metals side first and make sure we take into account the old values. This change is more in line with the current state.

Reply:
IMO it'll be better to set up proper enum on server side with fallback to old boolean behavior. I'm not even talking about changing names "false, "true", "minimal" but to have proper ADT on server side to avoid code like `userConfig().showInferredType.contains("true")` bur rather write `userConfig().showInferredType.contains(ShowInferredTypes.True)`